AsyncDisplayKit/ASViewController.mm

```diff
@@ -132,4 +142,70 @@ - (ASInterfaceState)interfaceState
   return _node.interfaceState;
 }
 
+#pragma mark - ASEnvironmentTraitCollection
+
+- (ASEnvironmentTraitCollection)displayTraitsForTraitCollection:(UITraitCollection *)traitCollection
+{
+  if (self.overrideDisplayTraitsWithTraitCollection) {
+    ASTraitCollection *asyncTraitCollection = self.overrideDisplayTraitsWithTraitCollection(traitCollection);
+    asyncTraitCollection.isMutable = NO;
+    return [asyncTraitCollection environmentTraitCollection];
+  }
+  
+  ASEnvironmentTraitCollection asyncTraitCollection = ASEnvironmentTraitCollectionFromUITraitCollection(traitCollection);
+  asyncTraitCollection.displayContext = _displayTraitsContext;
+  return asyncTraitCollection;
+}
+
+- (ASEnvironmentTraitCollection)displayTraitsForWindowSize:(CGSize)windowSize
+{
+  if (self.overrideDisplayTraitsWithWindowSize) {
+    ASTraitCollection *traitCollection = self.overrideDisplayTraitsWithWindowSize(windowSize);
+    traitCollection.isMutable = NO;
+    return [traitCollection environmentTraitCollection];
+  }
+  return self.node.environmentState.traitCollection;
+}
+
+- (void)progagateNewDisplayTraits:(ASEnvironmentTraitCollection)traitCollection
```

Comment:
The thinking was that the VC would be the one that knows of any special trait collection it may need based on the circumstances. Like UIViewController has  setOverrideTraitCollection method, not UIView.